sympy/logic/boolalg.py

```diff
@@ -3118,3 +3122,51 @@ def simplify_patterns_xor():
                       And(Lt(a, Max(b, c)), Ge(a, Min(b, c)))),
                      )
     return _matchers_xor
+
+
+# Classes corresponding to logic gates
+# Used in gateinputcount method
+BooleanGates = (And, Or, Xor, Nand, Nor, Not, Xnor, ITE)
```

Comment:
Maybe the supported gate types should be listed in the docstring. Could the existing BooleanFunction class be used here instead?

Reply:
Ideally it would be like that, but I do not think that `Implies`, `Equivalent`, and `Exclusive` should be in there, as there are typically no "gates" that directly implements them (one can of course build gates, but they are not really considered standard and for some of them, the internal complexity is pretty high). This sort of holds for `ITE` as well, a 2-1 multiplexer is more complicated than, say, a three-input `And`-gate, but that is still a very commonly used gate. (`Xor` is also more complicated.)

But it is a quite crude measure...

And, yes, it should be mentioned in the docstring which functions counts as gates.